toolchain/install/installation_test.cpp

```diff
@@ -0,0 +1,126 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/install/installation.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "common/ostream.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "testing/base/gtest_main.h"
+
+namespace Carbon {
+namespace {
+
+using ::testing::HasSubstr;
+using ::testing::NotNull;
+using ::testing::StrEq;
+using ::testing::StrNe;
+
+// Compute the test runfiles path, either using Bazel's `TEST_SRCDIR`
+// environment variable or the name of the test executable. Trying both of these
+// allows tests to be run directly out of the Bazel build tree, for example by a
+// debugger, and still find their runfiles.
+//
+// TODO: Extract this to a testing helper library and use it more broadly.
+static auto FindTestRunfiles() -> std::string {
+  std::string runfiles_path = getenv("TEST_SRCDIR");
```

Comment:
Note, see above comment about runfiles. I think we should discuss what's awkward, because we do need to accommodate various test execution environments.